File: blaze_flutter_sdk/android/src/main/kotlin/com/blaze/flutterblazesdk/players/tabs/BlazeReactPlayerTabsStyleMerger.kt
package com.blaze.flutterblazesdk.players.tabs

import android.content.Context
import com.blaze.blazesdk.style.players.tabs.BlazePlayerTabItemStyle
import com.blaze.blazesdk.style.players.tabs.BlazePlayerTabsStyle
import com.blaze.blazesdk.style.shared.models.blazeDp
import com.blaze.flutterblazesdk.utils.parsing.mergedWith
import com.blaze.flutterblazesdk.utils.parsing.safeParseColor
import com.blaze.flutterblazesdk.utils.parsing.toFontResId

/**
 * Merges the Dart-supplied [BlazeReactPlayerTabsStyle] onto a native [BlazePlayerTabsStyle]
 * (typically `BlazePlayerTabsStyle.base()`). Every field is optional; omitted fields keep their
 * native default. Mirrors the player-style mergers in this package.
 */
fun BlazePlayerTabsStyle.mergedWith(
    customization: BlazeReactPlayerTabsStyle?,
    context: Context
): BlazePlayerTabsStyle {
    customization ?: return this

    val merged = this

    merged.padding.mergedWith(customization.padding)

    customization.gradient?.let { gradient ->
        gradient.isVisible?.let { merged.gradient.isVisible = it }
        safeParseColor(gradient.startColor)?.let { merged.gradient.startColor = it }
        safeParseColor(gradient.middleColor)?.let { merged.gradient.middleColor = it }
        safeParseColor(gradient.endColor)?.let { merged.gradient.endColor = it }
    }

    customization.icon?.let { icon ->
        merged.icon?.let { iconStyle ->
            safeParseColor(icon.iconTint)?.let { iconStyle.iconTint = it }
            icon.padding?.let { iconStyle.padding = it.toInt().blazeDp }
        }
    }

    merged.selectedTabState.mergedWith(customization.selectedTabState, context)
    merged.unselectedTabState.mergedWith(customization.unselectedTabState, context)

    return merged
}

private fun BlazePlayerTabItemStyle.mergedWith(
    customization: BlazeReactPlayerTabItemStyle?,
    context: Context
): BlazePlayerTabItemStyle {
    customization ?: return this

    val merged = this

    customization.font?.toFontResId(context)?.let { merged.fontResId = it }
    safeParseColor(customization.textColor)?.let { merged.textColor = it }
    customization.textSize?.let { merged.textSize = it }
    // Dart letterSpacing is in points (like iOS); Android's TextView expects em, so divide by textSize.
    customization.letterSpacing?.let { letterSpacingPt ->
        merged.letterSpacing = if (merged.textSize > 0f) letterSpacingPt / merged.textSize else letterSpacingPt
    }
    customization.lineHeight?.let { merged.lineHeight = it.toInt().blazeDp }

    return merged
}

File: blaze_flutter_sdk/ios/blaze_flutter_sdk/Sources/Players/Tabs/BlazeReactPlayerTabsStyleMerger.swift
//
//  BlazeReactPlayerTabsStyleMerger.swift
//  blaze_flutter_sdk
//
//  Merges a Dart-supplied `BlazeReactPlayerTabsStyle` onto a native `BlazePlayerTabsStyle`
//  (typically `.base()`). Every field is optional; omitted fields keep their native default.
//  Mirrors the player-style mergers.
//

import BlazeSDK
import Foundation
import UIKit

extension BlazePlayerTabsStyle {

    func mergedWith(_ customization: BlazeReactPlayerTabsStyle?) -> BlazePlayerTabsStyle {
        guard let customization else { return self }

        var merged = self

        merged.padding = merged.padding.mergedWith(customization.padding)

        if let gradient = customization.gradient {
            merged.gradient.isVisible = gradient.isVisible ?? merged.gradient.isVisible
            merged.gradient.startColor = UIColor(hexString: gradient.startColor) ?? merged.gradient.startColor
            merged.gradient.middleColor = UIColor(hexString: gradient.middleColor) ?? merged.gradient.middleColor
            merged.gradient.endColor = UIColor(hexString: gradient.endColor) ?? merged.gradient.endColor
        }

        if let icon = customization.icon, var iconStyle = merged.icon {
            iconStyle.iconTint = UIColor(hexString: icon.iconTint) ?? iconStyle.iconTint
            iconStyle.padding = icon.padding ?? iconStyle.padding
            merged.icon = iconStyle
        }

        merged.selectedTabState = merged.selectedTabState.mergedWith(customization.selectedTabState)
        merged.unselectedTabState = merged.unselectedTabState.mergedWith(customization.unselectedTabState)

        return merged
    }
}

extension BlazePlayerTabItemStyle {

    func mergedWith(_ customization: BlazeReactPlayerTabItemStyle?) -> BlazePlayerTabItemStyle {
        guard let customization else { return self }

        var merged = self

        merged.textColor = UIColor(hexString: customization.textColor) ?? merged.textColor
        merged.letterSpacing = customization.letterSpacing ?? merged.letterSpacing
        // iOS embeds the size inside `UIFont` (no standalone `textSize`, unlike Android), handled by `fontWith`.
        merged.font = merged.font.fontWith(customization.font, textSize: customization.textSize)

        return merged
    }
}

extension BlazeSDK.BlazePlayerTabItemCustomImageStates? {

    func mergedWith(_ customization: BlazeReactPlayerButtonCustomImageStates?) -> Self {
        guard let customization else { return self }

        let selectedImage = UIImage.imageWith(customization.selectedImage) ?? self?.selectedImage
        let unselectedImage = UIImage.imageWith(customization.unselectedImage) ?? self?.unselectedImage

        guard selectedImage != nil || unselectedImage != nil else { return self }

        return .init(selectedImage: selectedImage, unselectedImage: unselectedImage)
    }
}
